### PR TITLE
Add tdd-style npm command

### PR DESCRIPTION
**Pull request by YonatanKra**
While developing two PR's I found it hard to work in TDD style since there was no ability to run the tests in `watch` mode. This PR adds this ability.
- Added npm script `test:tdd`
- Added gulp `tdd` variable
- Use the `tdd` variable in karma.conf in order to set autoWatch and allow multiple runs
- Make sure no page reloads happen during tests

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @YonatanKra! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/CesiumGS/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/YonatanKra/cesium/blob/FEAT-allow-tdd-style-testing-with-watch/CONTRIBUTORS.md) file!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/FEAT-allow-tdd-style-testing-with-watch/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by kring** [comment]
Thanks @YonatanKra, this looks reasonable to me. But unfortunately it breaks the normal testing workflow most Cesium developers are using:

1. `npm run build`
2. `npm start`
3. Navigate to http://localhost:8080
4. Click Run Tests

When I do this (Chrome / Windows), I get an error in the console and the tests don't load:

```
spec-main.js:24 Uncaught TypeError: Cannot assign to read only property 'reload' of object '[object Location]'
    at spec-main.js:24
```

While you're in there, please merge in master and update CHANGES.md. 
Thanks!

**Comment by YonatanKra** [comment]
Done :)

**Comment by kring** [comment]
Thanks @YonatanKra. The old workflow works well, now. The new workflow works too, though I had to rename `karma.conf.js` to `karma.conf.cjs` to get it to work in Node 12. This was probably a pre-existing problem with Karma in Cesium, but easy to fix.

I'm not sure I understand your workflow, though. The Cesium tests take quite awhile to run, so re-running them all each time you change any source file seems super inefficient. Most of us just navigate to here http://localhost:8080/Specs/SpecRunner.html?category=none and then click the spec we're working on. Needing to click refresh each time is annoying but tolerable. Then we just run all the specs at the very end to make sure we haven't broken something elsewhere. Are you doing something to limit which specs are run? Would you mind writing down some notes about your workflow? At least put them in this PR, but ideally they'd go in the testing guide: https://github.com/CesiumGS/cesium/tree/master/Documentation/Contributors/TestingGuide

Thanks again!

File: gulpfile.cjs [hunk 1]
```diff
@@ -921,6 +921,7 @@ gulp.task("test", function (done) {
   var release = argv.release ? argv.release : false;
   var failTaskOnError = argv.failTaskOnError ? argv.failTaskOnError : false;
   var suppressPassed = argv.suppressPassed ? argv.suppressPassed : false;
+  global.tdd = argv.tdd ? argv.tdd : false;
```
Comment (mramato) [inline gulpfile.cjs:924]: I don't think setting this global is necessary, you should be able to just pass the options to `new Karma.Server(` and that overrides the defaults in karma.conf

Comment (YonatanKra) [reply gulpfile.cjs]: @mramato I changed both the coverage and the tests definitions.

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @YonatanKra! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @YonatanKra! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @YonatanKra! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @YonatanKra! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by cesium-concierge** [comment]
Thanks again for your contribution @YonatanKra! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

**Comment by hpinkos** [comment]
Closing due to inactivity

**Pull request closed without merging**